esp/src/eclwatch/TopologyWidget.js

```diff
@@ -82,65 +82,69 @@ define([
-                                    GetProcessorInfo:filter.GetProcessorInfo,
-                                    GetStorageInfo:filter.GetStorageInfo,
-                                    LocalFileSystemsOnly:filter.LocalFileSystemsOnly,
-                                    GetSoftwareInfo:filter.GetSoftwareInfo,
-                                    DiskThreshold:filter.DiskThreshold,
-                                    DiskThresholdType:filter.DiskThresholdType,
-                                    ApplyProcessFilter:filter.ApplyProcessFilter,
-                                    AddProcessesToFilter: filter.AddtionalProcessesToFilter,
-                                    cbAutoRefresh: filter.cbAutoRefresh,
-                                }
-                            }).then(function (response) {
-                                var pfTab = context.ensureMIPane(response.GetMachineInfoResponse.Machines.MachineInfoEx[0].Address, {
-                                    params: response.GetMachineInfoResponse
-                                });
-                                context.detailsWidget.requestInformationWidget.set("disabled", false);
-                                context.detailsWidget.requestInformationWidget.init(response.GetMachineInfoResponse);
-                                pfTab.init(response.GetMachineInfoResponse, "machines");
+                        var request = {
+                            Path: selection[i].__hpcc_treeItem.Path,
+                            Cluster: selection[i].__hpcc_treeItem.Name,
+                            AutoRefresh: filter.AutoRefresh,
+                            MemThreshold: filter.MemThreshold,
+                            CpuThreshold: filter.CpuThreshold,
+                            MemThresholdType: filter.MemThreshold,
+                            GetProcessorInfo: filter.GetProcessorInfo,
+                            GetStorageInfo: filter.GetStorageInfo,
+                            LocalFileSystemsOnly: filter.LocalFileSystemsOnly,
+                            GetSoftwareInfo: filter.GetSoftwareInfo,
+                            DiskThreshold: filter.DiskThreshold,
+                            DiskThresholdType: filter.DiskThresholdType,
+                            ApplyProcessFilter: filter.ApplyProcessFilter,
+                            AddProcessesToFilter: filter.AddtionalProcessesToFilter,
+                            cbAutoRefresh: filter.cbAutoRefresh
+                        };
+                        request[MachineInformationClean] = filter[MachineInformationClean];
+                        request["Addresses.itemcount"] = MachineInformationCount;
+                        WsMachine.GetMachineInfo({
+                            request: request
+                        }).then(function (response) {
+                            var pfTab = context.ensureMIPane(response.GetMachineInfoResponse.Machines.MachineInfoEx[0].Address, {
```

Comment:
Although, not touched. This ensurePane could potentially need another unique id component to make it truly unique. Which is what is causing not all tabs to open.